sdk/inc/azure/core/az_json.h

```diff
@@ -88,7 +88,9 @@ AZ_NODISCARD az_result az_json_token_get_boolean(az_json_token const* json_token
  * @param json_token A pointer to an #az_json_token instance.
  * @param out_value A pointer to a variable to receive the value.
  * @return AZ_OK if the number is returned.<br>
- * AZ_ERROR_JSON_INVALID_STATE if the kind != AZ_JSON_TOKEN_NUMBER.
+ * AZ_ERROR_JSON_INVALID_STATE if the kind != AZ_JSON_TOKEN_NUMBER.<br>
+ * AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit is found within the token or if it contains
```

Comment:
> Is the JSON token valid if it has an invalid char?

No, it isn't, and the JSON token that the parser returns won't contain invalid characters for a number.

However, the token doesn't have to be invalid for us to return this az_result. It could be that the token contains some value that overflows an int32 (For example, a JSON number: 99999999999). I could remove the `_PARSER_` infix from that az_result to make it more general purpose.

AZ_ERROR_PARSER_UNEXPECTED_CHAR -> AZ_ERROR_UNEXPECTED_CHAR

This is the `az_result` that `az_span_atoi32(...)` returns

Reply:
Since such a rename has impacts across the repo, outside of the scope of this PR, I will address this separately.
https://github.com/Azure/azure-sdk-for-c/issues/905